tests/pyoozie/test_client.py

```diff
@@ -399,8 +399,38 @@ def test_jobs_query_coordinator_pagination(self, _, api):
             mock_get.side_effect = lambda url: next(mock_results)
             result = api._jobs_query(model.ArtifactType.Coordinator)
             assert len(result) == 3
-            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=1&len=500')
-            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=501&len=500')
+            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=1&len=5000')
+            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=5001&len=5000')
+            with pytest.raises(StopIteration):
+                next(mock_results)
+
+    @mock.patch.object(model.Coordinator, 'fill_in_details', side_effect=lambda c: c, autospec=True)
+    def test_jobs_query_coordinator_limit(self, _, api):
+        # mock_result = {'total': 1, 'coordinatorjobs': [{'coordJobId': '3-C'}]}
+        mock_results = iter(
+            [
+                {
+                    'total': 2,
+                    'coordinatorjobs': [{'coordJobId': '1-C'}, {'coordJobId': '2-C'}]
+                },
+                {
+                    'total': 5001,
+                    'coordinatorjobs': [{'coordJobId': '1-C'}, {'coordJobId': '2-C'}]
+                },
+                {
+                    'total': 5001,
+                    'coordinatorjobs': [{'coordJobId': '3-C'}]
+                }
+            ]
+        )
+
+        with mock.patch.object(api, '_get') as mock_get:
+            mock_get.side_effect = lambda url: next(mock_results)
+            api._jobs_query(model.ArtifactType.Coordinator, limit=5)
+            mock_get.assert_called_with('jobs?jobtype=coordinator&offset=1&len=5')
+            api._jobs_query(model.ArtifactType.Coordinator, limit=6000)
+            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=1&len=5000')
+            mock_get.assert_any_call('jobs?jobtype=coordinator&offset=5001&len=5000')
```

Comment:
Is this test substantially different from / combinable with `test_jobs_query_coordinator_pagination`?
https://github.com/Shopify/pyoozie/blob/045fe6f04ec7f5abbae1a8567feb1fb7bb87ddd4/tests/pyoozie/test_client.py#L383-L405

Reply:
The difference is testing pagination with and without an explicit limit.  It's pretty subtle but the previous version of the limit code would effectively disable pagination by setting the chunk to be equal to the limit.  This meant that if you set a limit of 20k then you'll request one chunk of 20k instead of X chunks of 500 up to the limit of 20k total results.

[The change that I'm introducing](https://github.com/Shopify/pyoozie/pull/72/files#diff-474ab6c8f76e0a89751359f5b7e19f1dR242) will make sure that even a large limit will be paginated.  This test checks for that by setting a limit of `6000` and checking that two REST calls were made with `len=5000` (the new chunk size).